punishments.js

```diff
@@ -126,14 +172,65 @@ Punishments.savePunishments = function () {
 	fs.writeFile(PUNISHMENT_FILE, buf, () => {});
 };
 
+Punishments.saveRoombans = function () {
+	let saveTable = new Map();
+	Punishments.roombannedIps.forEach((punishment, ip) => {
+		if (Date.now() >= punishment[2]) {
+			Punishments.roombannedIps.delete(ip);
+			return;
+		}
+		let id = punishment[1];
+		if (id.charAt(0) === '#') return;
+		let entry = saveTable.get(id);
+
+		if (entry) {
+			entry.keys.push(ip);
+			return;
+		}
+
+		entry = {
+			keys: [ip],
+			punishType: punishment[0],
+			rest: punishment.slice(2),
+		};
+		saveTable.set(id, entry);
+	});
+	Punishments.roombannedUserids.forEach((punishment, userid) => {
+		if (Date.now() >= punishment[2]) {
+			Punishments.roombannedUserids.delete(userid);
+			return;
+		}
+		let id = punishment[1];
+		let entry = saveTable.get(id);
+
+		if (!entry) {
+			entry = {
+				keys: [],
+				punishType: punishment[0],
+				rest: punishment.slice(2),
+			};
+			saveTable.set(id, entry);
+		}
+
+		if (userid !== id) entry.keys.push(userid);
+	});
+
+	let buf = 'Punishment\tUser ID\tRoom Id | IPs and alts\tExpires\r\n';
```

Comment:
Not as in "that'd be better", as in "that's what your code actually does, you just didn't realize it"